sematic/config/settings.py

```diff
@@ -138,6 +138,25 @@ def get_active_settings() -> ProfileSettings:
     return _ACTIVE_SETTINGS
 
 
+T = TypeVar("T", bound=Type[AbstractPlugin])
+
+
+def get_plugins_with_interface(
+    interface: T, scope: PluginScope, default: List[Type[AbstractPlugin]]
```

Comment:
I think this should work:
```suggestion
    interface: T, scope: PluginScope, default: List[T]
```
You can also drop the cast on `default` below with this change.

Reply:
I actually tried that originally and it didn't work; it claimed it was unable to infer `T` when I did it